src/Tgstation.Server.Host/Configuration/ElasticsearchConfiguration.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace Tgstation.Server.Host.Configuration
+	{
+	/// <summary>
+	/// Configuration options pertaining to elasticsearch log storage.
+	/// </summary>
+	sealed class ElasticsearchConfiguration
+	{
+		/// <summary>
+		/// The key for the <see cref="Microsoft.Extensions.Configuration.IConfigurationSection"/> the <see cref="ElasticsearchConfiguration"/> resides in.
+		/// </summary>
+		public const string Section = "Elasticsearch";
+
+		/// <summary>
+		/// Default value of <see cref="Enable"/>.
+		/// </summary>
+		const bool DefaultEnable = false;
+
+		/// <summary>
+		/// Default value of <see cref="Host"/>.
+		/// </summary>
+		const string DefaultHost = "http://127.0.0.1:9200"; // localhost
+
+		/// <summary>
+		/// Default value of <see cref="Username"/>.
+		/// </summary>
+		const string DefaultUsername = "my_username";
+
+		/// <summary>
+		/// Default value of <see cref="Password"/>.
+		/// </summary>
+		const string DefaultPassword = "my_password";
+
+		/// <summary>
+		/// Do we want to enable elasticsearch or not?.
+		/// </summary>
+		public bool Enable { get; set; } = DefaultEnable;
+
+		/// <summary>
+		/// The host of the elasticsearch endpoint.
+		/// </summary>
+		public string Host { get; set; } = DefaultHost;
```

Comment:
```suggestion
		public Uri Host { get; set; } = DefaultHost;
```

Reply:
Does not work as the default value needs to be a string unless I instance it 
```cs
		/// <summary>
		/// The host of the elasticsearch endpoint.
		/// </summary>
		public Uri Host { get; set; } = new Uri(DefaultHost);
```
And this feels too clunky, though ill do it if needed